download/download.go

```diff
@@ -26,6 +26,21 @@ const (
 	minRetryDelay = time.Millisecond * 100
 )
 
+// LogConfig defines a set of configuration parameters for downloader logging
+// calls.
+type LogConfig struct {
+	Attributes [][2]string
+}
+
+// Interface can be implemented to download via arbitrary transports/formats.
+type Interface interface {
+	WithLogConfig(LogConfig) Interface                    // set log configuration parameters
```

Comment:
We can probably omit the `WithLogConfig` and `LogConfig` part of the interface, a quick check with the github search doesn't show the older `WithLogAttrs` being used anywhere.

I'd vote to keep it on the standard http downloader so that anyone using one directly could still use it, for backwards compatibility, but leave it out of the new interface until we need it (unless we already do and I'm missing it)

Reply:
I think the log attributes/configuration are useful for correlating rest client debug messages with the plugin that owns the rest client. I also thought about removing it but while debugging a recent issue, I noticed we're not passing the log attributes down and it was a bit annoying to tell which plugin was making the call.

Reply:
>  I noticed we're not passing the log attributes down and it was a bit annoying to tell which plugin was making the call.

Ahh yea, if we fix that I'm definitely on board with keeping it.

Reply:
It's easy enough to fix and this would be a bit more future proof (struct instead of slice). If you're onboard, I'll fixup the PR and merge then?

Reply:
Yea, lets do it.